bin/BuildPackages.sh

```diff
@@ -60,36 +30,51 @@ then
         "Type: cd ../pkg; ../bin/BuildPackages.sh"
 fi
 
-# CURDIR
 CURDIR="$(pwd)"
+GAPDIR="$(cd .. && pwd)"
+COLORS=yes
+PACKAGES=
+
+# If output does not go into a terminal (but rather into a log file),
+# turn of colors.
+[[ -t 1 ]] || COLORS=no
+
+while [[ "$#" -ge 1 ]]; do
+  option="$1" ; shift
+  case "$option" in
+    --with-gaproot)   GAPDIR="$1"; shift ;;
+    --with-gaproot=*) GAPDIR=${option#--with-gaproot=}; shift ;;
+    --no-color)       COLORS=no ;;
+    *)                PACKAGES="$PACKAGES $option" ;;
+  esac
+done
 
-# GAPDIR
-GAPDIR=""
-if [[ "$#" -ge 1 ]]
+# packages to build
+if [[ -z "$PACKAGES" ]]
 then
-  case "$1" in
-    --with-gaproot=*)
-      GAPDIR=$(sed 's|--with-gaproot=||' <<< "$1")
-      shift
-    ;;
-
-    *)
-      cd ..
-      GAPDIR="$(pwd)"
-      cd "$CURDIR"
-    ;;
-  esac
+  PACKAGES="$(find . -maxdepth 2 -type f -name PackageInfo.g  | xargs -n 1 dirname)"
```

Comment:
The code using nullglob does not give all directories, it gives *all files*. But we only want directories that contain `PackageInfo.g`. 

And I still want to do the later check -- but now I can use it to valide that user arguments: I.e. if the user accidentally specified an invalid package directory name, we can show a warning (I actually forgot to add this, but added it now -- thanks for the poke).
Note that I really want this, because it actually bit me before (e.g. I entered `$GAPROOT/bin/BuildPackages.sh io orb` and wondered that nothing happened)